File: gnc/ctl_mpc/acados/external/hpipm/experimental/andrea/interfaces/julia/hpipm_solver.jl
type hpipm_solver
    # "attribtues"
    py_hpipm_solver
    py_hpipm_data
    py_hpipm_dims

    # "methods"
    solve
    print_sol

    function hpipm_solver(ju_qp_dims, ju_qp_data)
        this = new()

        py_qp_data = hp.hpipm_data()
        this.py_hpipm_data = ju2py_obj(ju_qp_data, py_qp_data) 

        py_qp_dims = hp.hpipm_dims()
        this.py_hpipm_dims = ju2py_obj(ju_qp_dims, py_qp_dims) 

        py_qp_solver = hp.hpipm_solver(this.py_hpipm_dims, this.py_hpipm_data)
        this.py_hpipm_solver = py_qp_solver

        this.solve = function()
            (this.py_hpipm_solver)[:solve]()
        end

        this.print_sol = function()
            this.py_hpipm_solver[:print_sol]()
        end

        return this
    end
end

File: gnc/ctl_mpc/acados/external/hpipm/interfaces/archive/matlab/hpipm_matlab/+hpipm_matlab/archive/hpipm_solver.m
classdef hpipm_solver
   properties
	  py_hpipm_solver
	  py_qp_data
	  py_qp_dims
   end
   
   methods
	   function obj = hpipm_solver(m_qp_dims, m_qp_data)
			import hpipm_matlab.*
			hp = py.importlib.import_module('hpipm_python');
			py_qp_dims = hp.hpipm_dims();
			obj.py_qp_dims = m2py_obj(m_qp_dims, py_qp_dims);
			
			py_qp_data = hp.hpipm_data();
			obj.py_qp_data = m2py_obj(m_qp_data, py_qp_data);
			obj.py_hpipm_solver = hp.hpipm_solver(py_qp_dims, py_qp_data);		   
	   end
	   
	   function return_flag = solve(obj)
		   return_flag = int64(obj.py_hpipm_solver.solve().real);
	   end

	   function print_sol(obj)
		   obj.py_hpipm_solver.print_sol()
	   end
   end
end

